### PR TITLE
Feat/search

### PR DESCRIPTION
**Pull request by abhij1607**
This PR consist of 

- Search functionality
- search result page

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ function App() {
         <Route path="/" element={<AppLayout />}>
           <Route path="/" element={<Home />} />
           <Route path="/watch/:watchid" element={<SingleVideoPage />} />
+          <Route path="results" element={<SearchResult />} />
```
Comment (Hamdrive) [inline src/App.js]: route can be written as `/results` to differentiate as subroute

**Review by Hamdrive — COMMENTED** [review]
Good work implementing search functionality for videos
Code is clean and approach is good, along with video views

File: src/App.js (hunk 1 above)
Comment (abhij1607) [reply src/App.js]: thanks for pointing, I'll correct it

File: src/components/Navbar/SearchBox/SearchBox.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const SearchBox = () => {
+  const [searchKeyword, setSearchKeyword] = useState("");
+  const {
+    dataState: { videos },
+  } = useData();
+
+  const navigate = useNavigate();
+
+  const searchSuggestions = () => {
+    if (searchKeyword !== "") {
```
Comment (prachi-sahani) [inline src/components/Navbar/SearchBox/SearchBox.jsx:15]: can be written as `if(!searchKeyword)`

**Review by prachi-sahani — COMMENTED** [review]
implementation for "search" functionality looks good
clean code
good to merge

File: src/components/Navbar/SearchBox/SearchBox.jsx (hunk 2 above)
Comment (abhij1607) [reply src/components/Navbar/SearchBox/SearchBox.jsx:15]: thanks for the suggestion

**Pull request merged**